docs/reference/adaptors.rst

```diff
@@ -383,6 +383,40 @@ Adaptors
         * :func:`flux::scan`
         * :func:`flux::fold_first`
 
+``set_union``
+^^^^^^^^^^^^^
+
+..  function::
+    template <sequence Seq1, sequence Seq2, typename Cmp = std::ranges::less> \
+        requires compatible<Seq1, Seq2> and \
+                 strict_weak_order_for<Cmp, Seq1> and \
+                 strict_weak_order_for<Cmp, Seq2> \
```

Comment:
Because `set_op_compatible` is a non-public concept, I think it would be better to just say `requires see_below`, and then add a `:requires:` section underneath to explain what the requirements are -- I've done this in the documentation of a couple of the algorithms, where the requires clause was a bit complicated to list otherwise.

Reply:
Moved to `:requires:` section underneath, but I am not sure if the explanation is correct.